src/calendar-heatmap.js

```diff
@@ -62,14 +91,40 @@ function calendarHeatmap() {
     return chart;
   };
 
-  function chart() {
+  chart.onMouseOver = function (value) {
+    if (!arguments.length) { return onMouseOver(); }
+    onMouseOver = value;
+    return chart;
+  };
 
+  chart.onMouseOut = function (value) {
+    if (!arguments.length) { return onMouseOut(); }
+    onMouseOut = value;
+    return chart;
+  };
+
+  function chart() {
     d3.select(chart.selector()).selectAll('svg.calendar-heatmap').remove(); // remove the existing chart, if it exists
+    if(!chart.dateRange()) {
+      dateRange = d3.time.days(yearAgo, now); // generates an array of date objects within the specified range  
+    }
 
-    var dateRange = d3.time.days(yearAgo, now); // generates an array of date objects within the specified range
-    var monthRange = d3.time.months(moment(yearAgo).startOf('month').toDate(), now); // it ignores the first month if the 1st date is after the start of the month
+    // remove data outside the date range
+    var start = dateRange[0];
+    var end = dateRange[dateRange.length - 1];
+    function withinRange(d) {
+      return (d.date > start || d.date.getTime() === start.getTime()) && 
+      (d.date < end || d.date.getTime() === end.getTime());
+    }
+    data = data.filter(withinRange);
```

Comment:
An anonymous function is preferred in such cases, since you are not using it anywhere else

```javascript
data = data.filter(function (d) {
  return (d.date > start || d.date.getTime() === start.getTime()) && 
  (d.date < end || d.date.getTime() === end.getTime());
});
```